http/document_service.go

```diff
@@ -260,46 +270,145 @@ func decodeGetDocumentsRequest(ctx context.Context, r *http.Request) (*getDocume
 	}, nil
 }
 
-// handleGetDocument is the HTTP handler for the GET /api/v2/documents/:ns/:id route.
-func (h *DocumentHandler) handleGetDocument(w http.ResponseWriter, r *http.Request) {
+func (h *DocumentHandler) handlePostDocumentLabel(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
+	d, _, s, a, err := h.getDocument(w, r)
+	if err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+	req, err := decodePostLabelMappingRequest(ctx, r, influxdb.DocumentsResourceType)
+	if err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
 
-	req, err := decodeGetDocumentRequest(ctx, r)
+	if err := req.Mapping.Validate(); err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+
+	if err := h.LabelService.CreateLabelMapping(ctx, &req.Mapping); err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+
+	label, err := h.LabelService.FindLabelByID(ctx, req.Mapping.LabelID)
 	if err != nil {
 		EncodeError(ctx, err, w)
 		return
 	}
+	d.Labels = append(d.Labels, label)
+	if err := s.UpdateDocument(ctx, d, influxdb.Authorized(a)); err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+	if err := encodeResponse(ctx, w, http.StatusCreated, newLabelResponse(label)); err != nil {
+		logEncodingError(h.Logger, r, err)
+		return
+	}
+}
 
-	s, err := h.DocumentService.FindDocumentStore(ctx, req.Namespace)
+// handleDeleteDocumentLabel will first remove the label from the document,
+// then remove that label.
+func (h *DocumentHandler) handleDeleteDocumentLabel(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	req, err := decodeDeleteLabelMappingRequest(ctx, r)
+	if err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+	d, _, s, a, err := h.getDocument(w, r)
 	if err != nil {
 		EncodeError(ctx, err, w)
 		return
 	}
 
-	a, err := pcontext.GetAuthorizer(ctx)
+	_, err = h.LabelService.FindLabelByID(ctx, req.LabelID)
 	if err != nil {
 		EncodeError(ctx, err, w)
 		return
 	}
+	// remove label from document
+	for k, l := range d.Labels {
+		if req.LabelID == l.ID {
+			d.Labels = append(d.Labels[:k], d.Labels[k+1:]...)
+			break
+		}
+	}
+
+	if err := s.UpdateDocument(ctx, d, influxdb.Authorized(a)); err != nil {
```

Comment:
See above comment. The document does not need to be updated; labels aren't stored within it.